### PR TITLE
IEN-348 Applicant data extract and IEN-435 Report 8 APIs

### PR DESCRIPTION
**Pull request by bhumin-fw**

### CODE REVIEW

**Review by coryatwater-FW — APPROVED** [review]
one comment, but looks good to me

File: apps/api/src/common/util.ts [hunk 1]
```diff
@@ -62,3 +62,12 @@ function getDateOfWeek(w: number, y: number) {
   else ISOweekStart.setDate(simple.getDate() + 8 - simple.getDay());
   return ISOweekStart;
 }
+
+export function startDateOfFiscal(dt: string) {
+  const today = new Date(dt);
+  let fiscalyearstart = today.getFullYear();
```
[the remark this answers is not in the document]
Comment (sunghwan-park-fw) [reply apps/api/src/common/util.ts:68]: Maybe it's time to think about our naming rule.

[the remark this answers is not in the document]
Comment (bhumin-fw) [reply apps/api/src/common/util.ts:68]: camelcase makes sense in ts/js code. And design or identify rules for string, float, number, boolean, and function.

File: apps/api/src/report/report.controller.ts [hunk 2]
```diff
@@ -123,4 +123,34 @@ export class ReportController {
...
+    summary: 'Report 8: Number of Internationally Educated Nurse Registrants Working in BC',
+  })
+  @Get('/applicant/ha-current-period-fiscal')
+  @RouteAcceptsRoles(
+    ValidRoles.HEALTH_AUTHORITY,
+    ValidRoles.HEALTH_MATCH,
+    ValidRoles.MINISTRY_OF_HEALTH,
+  )
+  async getApplicantHAForCurrentPeriodFiscal(@Query('to') to: string): Promise<object[]> {
+    this.logger.log(`Report: Number of Internationally Educated Nurse Registrants Working in BC`);
```
Comment (sunghwan-park-fw) [inline apps/api/src/report/report.controller.ts:137]: Could we use interceptor or middleware for these loggings?

Comment (bhumin-fw) [reply apps/api/src/report/report.controller.ts:137]: @sunghwan-park-fw 
What do you suggest?
Like add a middleware that log http request(url, query param, etc)

Comment (sunghwan-park-fw) [reply apps/api/src/report/report.controller.ts:137]: Yes, including response logging.

Comment (bhumin-fw) [reply apps/api/src/report/report.controller.ts:137]: @sunghwan-park-fw 
Let me add a separate PR for the same. will that be fine?

**Pull request merged**